Tests/Add-JiraIssueWorklog.Tests.ps1

```diff
@@ -0,0 +1,120 @@
+            Write-Output $jiraServer
+        }
+
+        Mock Get-JiraIssue -ModuleName PSJira {
+            [PSCustomObject] @{
+                ID = $issueID;
+                Key = $issueKey;
+                RestUrl = "$jiraServer/rest/api/latest/issue/$issueID";
+            }
+        }
+
+        Mock Invoke-JiraMethod -ModuleName PSJira -ParameterFilter {$Method -eq 'POST' -and $URI -eq "$jiraServer/rest/api/latest/issue/$issueID/worklog"} {
+            if ($ShowMockData)
+            {
+                Write-Host "       Mocked Invoke-JiraMethod with POST method" -ForegroundColor Cyan
+                Write-Host "         [Method] $Method" -ForegroundColor Cyan
+                Write-Host "         [URI]    $URI" -ForegroundColor Cyan
+            }
+
+            # This data was created from a GUI REST client, then sanitized. A lot of extra data was also removed to save space.
+            # Many Bothans died to bring us this information.
+            ConvertFrom-Json2 $restResponse
+        }
+
+        # Generic catch-all. This will throw an exception if we forgot to mock something.
+        Mock Invoke-JiraMethod -ModuleName PSJira {
+            Write-Host "       Mocked Invoke-JiraMethod with no parameter filter." -ForegroundColor DarkRed
+            Write-Host "         [Method]         $Method" -ForegroundColor DarkRed
+            Write-Host "         [URI]            $URI" -ForegroundColor DarkRed
+            throw "Unidentified call to Invoke-JiraMethod"
+        }
+
+        #############
+        # Tests
+        #############
+
+        It "Adds a comment to an issue in JIRA" {
+            $commentResult = Add-JiraIssueWorklog -Comment 'This is a test worklog entry from Pester.' -Issue $issueKey -TimeSpent 3600 -DateStarted "2018-01-01"
+            $commentResult | Should Not BeNullOrEmpty
+
+            # Get-JiraIssue should be used to identify the issue parameter
+            Assert-MockCalled -CommandName Get-JiraIssue -ModuleName PSJira -Exactly -Times 1 -Scope It
+
+            # Invoke-JiraMethod should be used to add the comment
+            Assert-MockCalled -CommandName Invoke-JiraMethod -ModuleName PSJira -Exactly -Times 1 -Scope It
+        }
+
+        It "Accepts pipeline input from Get-JiraIssue" {
+            $commentResult = Get-JiraIssue -InputObject $issueKey | Add-JiraIssueWorklog -Comment 'This is a test comment from Pester, using the pipeline!' -TimeSpent "3600" -DateStarted "2018-01-01"
+            $commentResult | Should Not BeNullOrEmpty
+
+            # Get-JiraIssue should be called once here, and once inside Add-JiraIssueWorklog (to identify the InputObject parameter)
+            Assert-MockCalled -CommandName Get-JiraIssue -ModuleName PSJira -Exactly -Times 2 -Scope It
+            Assert-MockCalled -CommandName Invoke-JiraMethod -ModuleName PSJira -Exactly -Times 1 -Scope It
+        }
+
+        It "Outputs the comment as a PSJira.Worklogitem object" {
```

Comment:
Testing the type of the output object is done in the `ConvertTo-Jira*.Tests.ps1` function.

here is an example: https://github.com/PSJira/PSJira/blob/master/Tests/ConvertTo-JiraComment.Tests.ps1#L62

_this will fix this test: https://ci.appveyor.com/project/JoshuaT/psjira/build/1.2.5.248#L83_